### PR TITLE
Snprintf2

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Comment by byrnHDF** [comment]
Why all the format changes?

**Comment by byrnHDF** [comment]
> Why all the format changes?

Oh because this was after the sprintf to snprintf changes.

**Review by byrnHDF — APPROVED** [review]

**Review by jhendersonHDF — APPROVED** [review]

File: test/fheap.c [hunk 1]
```diff
@@ -565,8 +565,9 @@ begin_test(fheap_test_param_t *tparam, const char *base_desc, fheap_heap_ids_t *
      */
     del_str = get_del_string(tparam);
     HDassert(del_str);
-    test_desc = (char *)H5MM_malloc(HDstrlen(del_str) + HDstrlen(base_desc));
-    HDsprintf(test_desc, base_desc, del_str);
+    size_t test_desc_len = HDstrlen(del_str) + HDstrlen(base_desc);
+    test_desc            = (char *)H5MM_malloc(test_desc_len);
+    HDsnprintf(test_desc, test_desc_len, base_desc, del_str);
```
Comment (gnuoyd) [inline test/fheap.c:570]: Something funny is happening here. I will come back to it.

Comment (gnuoyd) [reply test/fheap.c:570]: I fixed the funny business, see PR #1401.

**Review by gnuoyd — APPROVED** [review]
LGTM.

**Pull request merged**